scripts/setup-emlinux

```diff
@@ -1,32 +1,62 @@
 #!/bin/bash
 
+clean_up()
+{
+    unset BUILDDIR REPOS SYNC_POKY POKY_REV
+}
+
+# save error during 'source setup-emlinux'
+_ERROR=0
+trap '_ERROR=$?' ERR
+
 BUILDDIR="build"
 if [ -n "$1" ]; then
     BUILDDIR=$1
 fi
 
-sudo apt-get update
-sudo apt-get install gawk wget git-core diffstat unzip texinfo gcc-multilib \
-     build-essential chrpath socat cpio python python3 python3-pip python3-pexpect \
-     xz-utils debianutils iputils-ping libsdl1.2-dev xterm
+REPOS=${PWD}/repos
+if [ ! -e ${REPOS}/poky ]; then
```

Comment:
The "-e" expression doesn't care about given path is file or directory or other type, so  "-d" would be nice to check directory exists or not.

Reply:
Yes. I will use "-d".